src/app/shared/services/api.service.ts

```diff
@@ -128,11 +126,18 @@ export class ApiService {
     );
   }
 
-  getPersonCast$(params: string): Observable<{ cast: CastCard[], crew: CastCard[] }> {
-    return this.http.get<{ cast: CastDTO[], crew: CastDTO[] }>(params).pipe(
-      map((data) => ({
-        cast: data.cast.map((castItem) => castMapper(castItem)),
-        crew: data.crew.map((castItem) => castMapper(castItem)),
+  getPersonCast$(params: string): Observable<CastAndCrew> {
+    return this.http.get<CastAndCrewDTO>(params).pipe(
+      map((data: CastAndCrewDTO) => ({
+        cast: data.cast
+          .map((castItem: CastDTO) => {
+            return castItem.title ? castMovieMapper(castItem) : castTvMapper(castItem);
```

Comment:
It looks like small hack everywhere.
It better to use guard type here, instead just property checking.

```
function isMovieCast(person: CastDTO): person is CastDTO {
  return cast.title. !== undefined;
}
```

But my code above doesn't make sense cause there is not difference between input and output type. It happens due to general CastDTO, this interface contains all properties of world. Let's divide it for two individual interfaces and after use union

```
type CastMovieDto = {
  title: string;
  ...
}

type CastTvDto = {
  name: string;
  ...
}

type CastDTO = CastMovieDto | CastTvDto;
```

Sorry, I didn't noticed that earlier.